### ISSUE TITLE
Support for NiftyReg

### ISSUE THREAD

**Issue by pwighton** [ReproNim/neurodocker#377]
Is it possible to add NiftyReg to the list of supported software?

The version available via neurodebian or via [binaries](https://sourceforge.net/projects/niftyreg/) is fairly old (v1.3.9; 2012).  It would be nice to be able to easily get a more modern version (v1.5.58; 2018) which is a pre-req for the infant pipeline in FreeSurfer.

The code at https://git.code.sf.net/p/niftyreg/git compiles cleanly in `pwighton/fs-dev-build` ([dockerfile](https://github.com/pwighton/fs-docker/blob/master/build/Dockerfile))

**Comment by pwighton** [comment]
Actually, it looks like the source now lives at https://github.com/KCL-BMEIS/niftyreg

**Comment by kaczmarj** [comment]
thanks @pwighton - definitely possible to do this. please let me finish the refactor in https://github.com/ReproNim/neurodocker/tree/refactor/migrate-to-reproenv first.

that refactor introduces a generic container generator called "reproenv". it will act as the "engine" of neurodocker. it also introduces a better spec and a cleaner code base.

**Comment by pwighton** [comment]
Thanks @kaczmarj, sounds good!

**Comment by kaczmarj** [comment]
i had a chance to work on  this. i'm not sure which repository is the main one (https://cmiclab.cs.ucl.ac.uk/mmodat/niftyreg or https://github.com/KCL-BMEIS/niftyreg). here is a dockerfile that works.

```Dockerfile
FROM debian:stretch-slim
RUN apt-get update \
    && apt-get install --yes --no-install-recommends \
        ca-certificates cmake gcc g++ git make \
    && git clone https://cmiclab.cs.ucl.ac.uk/mmodat/niftyreg.git /tmp/niftyreg-src \
    && mkdir /tmp/niftyreg-build \
    && cd /tmp/niftyreg-build \
    && cmake -DCMAKE_INSTALL_PREFIX=/opt/niftyreg /tmp/niftyreg-src \
    && make \
    && make install
ENV PATH=/opt/niftyreg/bin:$PATH
```

then build and run

```bash
docker build -t niftyreg - < niftyreg.Dockerfile
docker run --rm -it niftyreg reg_tools --help
```

this compiles static libraries. it's possible to use shared libraries, but that increases complexity of the dockerfile without much size benefits.

next step is to convert this to a neurodocker specification.

**Comment by pwighton** [comment]
Thanks @kaczmarj, I'll give that a shot and see how far I get.

**Comment by pwighton** [comment]
@kaczmarj, I made a PR with my initial attempt (#379).  Any suggestions?  Looks like I need to parameterize `version` somewhere?

**Comment by kaczmarj** [comment]
closed by #380
